pystac_client/client.py

```diff
@@ -195,21 +201,78 @@ def from_file(  # type: ignore
             )
 
         client: Client = super().from_file(href, stac_io)
-
-        client._stac_io._conformance = client.extra_fields.get(  # type: ignore
-            "conformsTo", []
-        )
         client.modifier = modifier
 
         return client
 
-    def _supports_collections(self) -> bool:
-        return self._conforms_to(ConformanceClasses.COLLECTIONS) or self._conforms_to(
-            ConformanceClasses.FEATURES
+    def has_conforms_to(self) -> bool:
+        return bool(self._stac_io and "conformsTo" in self.extra_fields)
```

Comment:
Can you do one of the following?
- Remove the `self._stac_io` from the check, or
- Add a comment explaining why the `self._stac_io` check is needed

Thanks!